tiledb/sm/query/reader_base.h

```diff
@@ -547,6 +655,109 @@ class ReaderBase : public StrategyBase {
+   *
+   * @param num_chunks Total number of chunks in a tile
+   * @param num_range_threads Total number of range threads.
+   * @param range_thread_idx Current range thread index.
+   * @return {min, max}
+   */
+  std::tuple<uint64_t, uint64_t> compute_chunk_min_max(
+      const uint64_t num_chunks,
+      const uint64_t num_range_threads,
+      const uint64_t thread_idx) const;
+
+  /**
+   * Reads the chunk data of all tile buffers and stores them in a data
+   * structure together with the offsets between them
+   *
+   * @param name The attribute/dimension the tile belong to.
+   * @param tile The offsets tile to be unfiltered.
+   * @param tile_chunk_data The tile/offsets tile chunk info, buffers and
+   * offsets
+   * @param unfiltered_tile_size The size of the unfiltered tile buffer
+   * @param tile_var_chunk_data The value tile chunk info, buffers and offsets
+   * @param unfiltered_tile_var_size The size of the unfiltered tile_var buffer
+   * @param tile_validity_chunk_data The validity tile chunk info, buffers and
+   * offsets
+   * @param unfiltered_tile_validity_size The size of the unfiltered tile
+   * validity buffer
+   * @return Status
+   */
+  Status load_tile_chunk_data(
+      const std::string& name,
+      ResultTile* const tile,
+      ChunkData* const tile_chunk_data,
+      uint64_t* const unfiltered_tile_size,
+      ChunkData* const tile_chunk_var_data,
+      uint64_t* const unfiltered_tile_var_size,
+      ChunkData* const tile_chunk_validity_data,
+      uint64_t* const unfiltered_tile_validity_size) const;
+
+  /**
+   * Reads the chunk data of a tile buffer and populates a chunk data structure
+   *
+   * @param tile The offsets tile to be unfiltered.
+   * @param tile_chunk_data The tile chunk info, buffers and offsets
+   * @return Status
+   */
+  std::tuple<Status, std::optional<uint64_t>> load_chunk_data(
+      Tile* const tile, ChunkData* chunk_data) const;
+
+  /**
+   * Unfilter a specific range of chunks in tile
+   *
+   * @param name The attribute/dimension the tile belong to.
+   * @param tile The offsets tile to be unfiltered.
+   * @param range_thread_idx Current range thread index.
+   * @param num_range_threads Total number of range threads.
+   * @param tile_chunk_data The tile chunk info, buffers and offsets
+   * @param tile_var_chunk_data The value tile chunk info, buffers and offsets
+   * @param tile_validity_chunk_data The validity tile chunk info, buffers and
+   * offsets
+   * @return Status
+   */
+  Status unfilter_tile_chunk_range(
+      const std::string& name,
+      ResultTile* const tile,
+      uint64_t range_thread_idx,
+      uint64_t num_range_threads,
+      const ChunkData& tile_chunk_data,
+      const ChunkData& tile_chunk_var_data,
+      const ChunkData& tile_chunk_validity_data) const;
+
+  /**
+   * Perform some necessary post-processing on a tile that was just unfiiltered
+   *
+   * @param name The attribute/dimension the tile belong to.
```

Comment:
Nit of all nits: I try not to use The as the first word of a param description as it's not really needed. Feel free to ignore 😅